samples/string_interning.cpp

```diff
@@ -0,0 +1,18 @@
+#include <vector>
+#include <string>
+
+char* add_string(char *string, std::vector<char*> &old_strings) {
+    //check if the string is stored
```

Comment:
`add_string` is probably not the best name in the world :) It's quite generic, I think something like `intern_string` is better.